backend/cmd/headlamp.go

```diff
@@ -424,6 +451,43 @@ func createHeadlampHandler(config *HeadlampConfig) http.Handler {
 		proxy.ServeHTTP(w, r)
 	})
 
+	// the endpoint removes all the cluster contexts created from the KubeConfig,
+	// and reloads the contexts from the KubeConfig file.
+	// This is useful when the KubeConfig file is updated.
+	r.HandleFunc("/config/clusters/refresh", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Can this be abused by blasting this endpoint (resulting in IO)?
An alternative would be to watch the directory for changes (we have an issue #322  ) but we have to think about the implications to the UX (browsing stuff while suddenly the config changes), so maybe one quick way to avoid this is to also pass the server a `--watch-kubeconfigs` option to listen to such changes only at will.

WDYT?
cc/ @illume / @ashu8912 / @yolossn

Reply:
I'm not sure about adding a --watch-kubeconfigs option. It would be useful for the Headlamp app in general to watch for kubeconfig changes(not just as a docker desktop extension), and I think @yolossn made an issue for that.

Auth for this URLs needs solving along with all of them when running as an app IMHO.

Reply:
Even if we choose to add a watcher the refresh endpoint does make sense, we can also extend this to let the user manually refresh the app. As far as abuse is concerned we have to take care of this across the whole app as @illume pointed out.

Reply:
I agree that limiting the scope of this feature would be good. The /refresh URL shouldn't be available for in-cluster deployments IMHO. Because I can't think of a use case where that makes sense.

Having this on ONLY when running in docker extension is probably where we should start. Because the other configurations will not make use of this feature yet. In future PRs they may, but they don't yet. If this is best accomplished by adding a `--watch-kubeconfigs`, then that's what I vote for.

Reply:
AFAIK there isn't a way for the backend to find if it is running as a docker extension, in frontend the checks have been added to load the docker setup page only when it is running as a docker extension. So the /refresh endpoint will not be called when headlamp is running in in-cluster mode.

Reply:
> Auth for this URLs needs solving along with all of them when running as an app IMHO.

Right. But so far we've been careful and more restrictive with the things we allow in-cluster.
One thing is a user having services who are abusing Headlamp's endpoints but somehow had to make it to the user's machine in the first place. Another thing is to have that be the case for a true server use-case (hosted Headlamp).

So I am still a bit reluctant in adding this endpoint. Of course we can mark it as experimental and avoid it in-cluster, but if we would like anyway to watch for kube config, why not just watch for it as part of this PR (or as a separate one before this)? (the frontend can already get the config which would indicate that something changed by comparison as a lower bar)

Slightly related, in case you can thing of other implications:
In the near future we will have a welcome screen where the user is supposed to select the clusters they want to use in Headlamp and it shall keep a copy of that context in Headlamp's own config (so we can manage it independently from the main kube configs). Maybe there's an option we can give to the user where they just keep using the default kube configs in which case watching for kube configs would still be necessary.